### PR TITLE
feat: update account

### PR DESCRIPTION
**Pull request by ebosetalee**
feat: get account details

### CODE REVIEW

File: src/routes/account.js [hunk 1]
```diff
@@ -2,11 +2,17 @@
...
-const { CREATE } = schema;
+const { CREATE, UPDATE } = schema;
 
 router.post("/", validate(CREATE), createAccount);
 
+router.get("/:id", getAccount);
+
+router.patch("/", validate(UPDATE), updateAccount);
+
+router.patch("/:id", updateAccountWithId);
```
Comment (jirevwe) [inline src/routes/account.js:16]: I think the validator middle-ware needs to be mounted here too
```suggestion
router.patch("/:id", validate(UPDATE), updateAccountWithId);
```

File: src/controllers/account.js [hunk 2]
```diff
@@ -23,8 +23,48 @@ export const createAccount = asyncHandler(async (req, res) => {
...
+export const updateAccountWithId = asyncHandler(async (req, res) => {
+    console.info("updating Account details using id...");
+
+    const account = await checkAccountExists(req.params.id, "_id");
+
+    if (!account) {
+        throw new ErrorResponse("Account not found", NOT_FOUND);
+    }
+
+    const data = await Account.findByIdAndUpdate(req.params.id, req.body, { new: true });
```
Comment (jirevwe) [inline src/controllers/account.js]: I think you might need to use `$set` here, just to follow best practices.
```suggestion
    const data = await Account.findByIdAndUpdate(req.params.id, $set: { req.body } , { new: true });
```

**Review by jirevwe — CHANGES_REQUESTED** [review]
Good job!!!

File: src/routes/account.js (hunk 1 above)
Comment (ebosetalee) [reply src/routes/account.js:16]: The validation requires account number 
but if the id is passed in the url is there a need for account number? @jirevwe

**Review by allengblack — APPROVED** [review]

**Review by jirevwe — APPROVED** [review]

**Pull request merged**